src/pages/settings/Preferences/LanguagesPage.js

```diff
@@ -0,0 +1,94 @@
+import ROUTES from '../../../ROUTES';
+import ONYXKEYS from '../../../ONYXKEYS';
+import OptionsList from '../../../components/OptionsList';
+import styles from '../../../styles/styles';
+import themeColors from '../../../styles/themes/default';
+import * as Expensicons from '../../../components/Icon/Expensicons';
+import * as App from '../../../libs/actions/App';
+
+const greenCheckmark = {src: Expensicons.Checkmark, color: themeColors.success};
+
+const propTypes = {
+    ...withLocalizePropTypes,
+};
+
+const LanguagesPage = (props) => {
+    const localesToLanguages = _.map(
+        [
+            {
+                value: 'en',
+                text: props.translate('languagesPage.languages.english'),
+            },
+            {
+                value: 'es',
+                text: props.translate('languagesPage.languages.spanish'),
+            },
+        ], language => (
+            {
+                ...language,
+                keyForList: language.value,
+
+                // Include the green checkmark icon to indicate the currently selected value
+                customIcon: props.preferredLocale === language.value ? greenCheckmark : undefined,
+
+                // This property will make the currently selected value have bold text
+                boldStyle: props.preferredLocale === language.value,
+            }
+        ),
+    );
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            <HeaderWithCloseButton
+                title={props.translate('languagesPage.language')}
+                shouldShowBackButton
+                onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_PREFERENCES)}
+                onCloseButtonPress={() => Navigation.dismissModal(true)}
+            />
+            <OptionsList
+                sections={[{data: localesToLanguages}]}
+                onSelectRow={
+                    (language) => {
+                        if (language.value !== props.preferredLocale) {
+                            App.setLocale(language.value);
+                        }
+                        Navigation.navigate(ROUTES.SETTINGS_PREFERENCES);
+                    }
+                }
+                hideSectionHeaders
+                optionHoveredStyle={
+                    {
+                        ...styles.hoveredComponentBG,
+                        ...styles.mln5,
+                        ...styles.mrn5,
+                        ...styles.pl5,
+                        ...styles.pr5,
+                    }
+                }
+                shouldHaveOptionSeparator
+                disableRowInnerPadding
+                contentContainerStyles={[styles.ph5]}
+            />
+        </ScreenWrapper>
+    );
+};
+
+LanguagesPage.displayName = 'LanguagesPage';
+LanguagesPage.propTypes = propTypes;
+
+export default compose(
+    withLocalize,
+    withOnyx({
+        preferredLocale: {
+            key: ONYXKEYS.NVP_PREFERRED_LOCALE,
+        },
```

Comment:
Huh, I see that the `preferredLocale` prop is defined in `localeProviderPropTypes` PropTypes (in `withLocalize`) but why don't we import `localeProviderPropTypes` in here and spread into `propTypes`?

I see we don't do that on other pages that subscribe to `key: ONYXKEYS.NVP_PREFERRED_LOCALE,` but I'm confused why 😅

Reply:
I think we do spread the props if the file is a UI component. 
Examples:
https://github.com/Expensify/App/blob/d5cfa894155283cb81ca5a17b6b623e534164ae7/src/components/LocalePicker.js#L25
https://github.com/Expensify/App/blob/d5cfa894155283cb81ca5a17b6b623e534164ae7/src/pages/home/sidebar/SidebarLinks.js#L73

But we don't do it in other stuff (e.g. anything in the `actions/` directory)

Reply:
True we spread `withLocalizePropTypes` but `preferredLocale` is defined in `localeProviderPropTypes` which we don't spread into components 🤔

Reply:
fixed.